src/redux/init.js

```diff
@@ -0,0 +1,48 @@
+const storedState = JSON.parse(localStorage.getItem("dataState"));
+const todoListsState = storedState.todoLists;
+
+export const getInitialState = () => {
+  Object.keys(todoListsState).forEach(todoListId => {
+    const todoList = todoListsState[todoListId];
+    const todos = todoList.todos;
+
+    Object.keys(todos).forEach(todoId => {
+      const todo = todos[todoId];
+      todo.dueDate = new Date(todo.dueDate);
+    })
+  })
+
+  const initialState = {
+    todoLists: todoListsState,
+    userPreferences: storedState.userPreferences,
+    notifications: { }
+  }
+
+  return initialState
+};
+
+export const afterStoreInitialization = store => {
+  setTimeout(() => triggerNotifications(store), 5000);
+};
+
+const triggerNotifications = store => {
+  setTimeout(() => {
```

Comment:
Ah, you use `setTimeout` twice. This means that the notifications will appear after 10s. This is not a big issue, but you probably want that

Reply:
no i don't :p